rskj-core/src/main/java/co/rsk/core/bc/BlockChainImpl.java

```diff
@@ -563,7 +563,7 @@ private void flushData() {
         return Block.getTxTrie(transactions).getHash();
     }
 
-    public static byte[] calcReceiptsTrie(List<TransactionReceipt> receipts) {
+    public synchronized static byte[] calcReceiptsTrie(List<TransactionReceipt> receipts) {
```

Comment:
Why add synchronized to a static method? which is the use case? The collection argument could mutate EVEN with the synchronized added to method. The variables are locals, and the trie is immutable

Reply:
It avoids writing multiple times on Trie hashes property by calling methods getHash( ) -> toMessage( ) -> getHash(K) -> setHash( )
Maybe we can get to a better solution for this.

Reply:
I don't still see the use case. The Trie hash property is only calculated once, at the en do f the method. I don't understand your reference to "-> toMessage" The above static method is only used in miner code. 

Out of scope: there is another private method in BlockResult. We should unify both. But it is out of scope of this pull request.

Reply:
I was confused about the static method, a new Trie is created everytime on the static method so there is no problem with the getHash method inside of calcReceiptsTrie. I'll remove the synchronized on this method. 

If getHash is called on another place with an already created Trie this problem could arise, if it's not precalculated to be stored inside of Trie it should use some kind of synchronization.